### PR TITLE
Documentation: note on the policy of insecure by default

### PR DESCRIPTION
**Pull request by mitake**
cc @xiang90 @spzala @gyuho

### CODE REVIEW

File: Documentation/op-guide/security.md [hunk 1]
```diff
@@ -426,6 +426,9 @@ Make sure to sign the certificates with a Subject Name the member's public IP ad
 
 The certificate needs to be signed for the member's FQDN in its Subject Name, use Subject Alternative Names (short IP SANs) to add the IP address. The `etcd-ca` tool provides `--domain=` option for its `new-cert` command, and openssl can make [it][alt-name] too.
 
+### etcd is insecure by default
+Note that etcd doesn't enforce users to use the authentication feature in the transport layer or [RBAC based authentication][auth]. It means an etcd cluster which doesn't enable the security features can expose its data to any clients. Users need to enable the features for avoiding such a situation.
```
Comment (spzala) [inline Documentation/op-guide/security.md:430]: @mitake thanks, this looks great but just a thought, if you and other reviewers like, -- how about putting this at the beginning of the document? Something like in bold in here "etcd supports automatic TLS as well as authentication through client certificates for both clients to server as well as peer (server to server / cluster) communication. **Note that these security features are not enabled by the default installation of etcd. To secure an etcd cluster and not expose its data to any clients, users nust enable one or more security features as described in this document.** "

Comment (philips) [reply Documentation/op-guide/security.md:430]: I think context on WHY etcd doesn’t have authentication, authorization, or TLS on by default is important as well: “etcd doesn’t enable auth or transport security by default to reduce friction for users getting started with the database. Further, changing this default would be a breaking change for the project which was established 7 years ago.”

Comment (spzala) [reply Documentation/op-guide/security.md:430]: @philips thanks for the insight, that's great point. Earlier "by design" was the only thing that we discussed for this doc update.

Comment (mitake) [reply Documentation/op-guide/security.md]: Thanks for your comments @spzala and @philips ! I updated this PR based on the comment. Could you take a look at the latest version?

File: Documentation/op-guide/security.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: Transport security model
 ---
 
-etcd supports automatic TLS as well as authentication through client certificates for both clients to server as well as peer (server to server / cluster) communication.
+etcd supports automatic TLS as well as authentication through client certificates for both clients to server as well as peer (server to server / cluster) communication. **Note that etcd doesn't enable [RBAC based authentication][auth] or the authentication feature in the transport layer by default to reduce friction for users getting started with the database. Further, changing this default would be a breaking change for the project which was established 7 years ago. An etcd cluster which doesn't enable the security features can expose its data to any clients. Users need to enable the features for avoiding such a situation.**
```
Comment (philips) [inline Documentation/op-guide/security.md:5]: s%the security features%security features%g

Personally I would just remove “ Users need to enable the features for avoiding such a situation” completely. I don’t think it adds anything to the warning.

Comment (mitake) [reply Documentation/op-guide/security.md]: Thanks for your comment. I updated based on your comment in the latest version, could you take a look again?

Comment (spzala) [reply Documentation/op-guide/security.md]: @mitake @philips I would also remove mentioning number of years. So I guess we can remove `which was established 7 years ago` altogether if that sounds good? Thanks!

Comment (philips) [reply Documentation/op-guide/security.md]: good point, just say since 2013 or soemthing.

**Comment by philips** [comment]
LGTM after 7 years -> since 2013 tweak.

**Comment by spzala** [comment]
Thanks @philips
LGTM too after that change @mitake Thanks!

File: Documentation/op-guide/security.md (hunk 2 above)
Comment (mitake) [reply Documentation/op-guide/security.md]: @spzala @philips yeah that makes sense, I updated that part in the latest commit. Thanks!

**Comment by philips** [comment]
lgtm after test failure investigation

**Comment by mitake** [comment]
@philips The failed test case is `TestCtlV3Lock` and it occasionally fails nondeterministically (e.g. https://github.com/etcd-io/etcd/issues/6464) because of the nature of etcd lock. I think we don't need to open a new issue for it.

**Comment by philips** [comment]
Ok sounds good

**Pull request merged**